src/OrbitGl/CaptureViewElementTest.cpp

```diff
@@ -67,4 +100,85 @@ TEST(CaptureViewElementTesterTest, PassesAllTestsOnExistingElement) {
 }
+
+const Viewport kViewport(100, 100);
+const TimeGraphLayout kLayout;
+
+TEST(CaptureViewElement, IsMouseOverWorksForVisibleElements) {
+  UnitTestCaptureViewLeafElement elem(nullptr, &kViewport, &kLayout);
+  elem.SetWidth(kViewport.GetWorldWidth());
+
+  EXPECT_FALSE(elem.IsMouseOver(Vec2(-1, -1)));
+  EXPECT_TRUE(elem.IsMouseOver(Vec2(5, 5)));
+  EXPECT_TRUE(elem.IsMouseOver(elem.GetSize()));
+  EXPECT_TRUE(elem.IsMouseOver(Vec2(0, 0)));
+}
+
+TEST(CaptureViewElement, IsMouseOverDoesNotWorkForElementsOutsideOfTheParent) {
+  const int kChildCount = 2;
+  UnitTestCaptureViewContainerElement elem(nullptr, &kViewport, &kLayout, kChildCount);
+
+  CaptureViewElement* child0 = elem.GetAllChildren()[0];
+
+  static_assert(kLeafElementHeight > 5);
+  child0->SetPos(0, -5);
+  EXPECT_TRUE(child0->IsMouseOver(Vec2(0, 0)));
+  EXPECT_FALSE(child0->IsMouseOver(Vec2(0, -1)));
+}
+
+TEST(CaptureViewElement, MouseWheelEventRecursesToCorrectChildren) {
+  using ::testing::_;
+  using ::testing::Exactly;
+  using ::testing::Return;
+
+  const int kChildCount = 2;
+  UnitTestCaptureViewContainerElement container_elem(nullptr, &kViewport, &kLayout, kChildCount);
+
+  static_assert(kMarginAfterChild > 0);
+
+  const Vec2 kPosOutside(-1, -1);
+  const Vec2 kPosOnParent(10, kLeafElementHeight + 1);
+  const Vec2 kPosOnChild0(10, kLeafElementHeight);
+  const Vec2 kPosOnChild1(10, kLeafElementHeight + kMarginAfterChild);
+  const Vec2 kPosOnChild2(10, (kLeafElementHeight + kMarginAfterChild) * 2);
+
+  class ElementThatHandlesMouseWheelEvents : public UnitTestCaptureViewLeafElement {
+   public:
+    explicit ElementThatHandlesMouseWheelEvents(CaptureViewElement* parent,
+                                                const Viewport* viewport,
+                                                const TimeGraphLayout* layout)
+        : UnitTestCaptureViewLeafElement(parent, viewport, layout) {}
+
+   protected:
+    [[nodiscard]] bool OnMouseWheel(const Vec2& mouse_pos, int delta,
+                                    const ModifierKeys& modifiers) override {
+      return true;
+    }
+  };
+
+  container_elem.AddChild(
+      std::make_unique<ElementThatHandlesMouseWheelEvents>(&container_elem, &kViewport, &kLayout));
+
+  const int kDelta = 1;
+
+  CaptureViewElementMock* child0 =
+      dynamic_cast<CaptureViewElementMock*>(container_elem.GetAllChildren()[0]);
+  CaptureViewElementMock* child1 =
+      dynamic_cast<CaptureViewElementMock*>(container_elem.GetAllChildren()[1]);
+
+  // Expect the parent to catch all mouse wheel events of children 0 and 1, but not those of child 2
+  // since child 2 actually handles the event
+  EXPECT_CALL(container_elem, OnMouseWheel(_, kDelta, _))
+      .Times(Exactly(3))
+      .WillRepeatedly(Return(false));
+
+  EXPECT_CALL(*child0, OnMouseWheel(_, kDelta, _)).Times(Exactly(1)).WillRepeatedly(Return(false));
+  EXPECT_CALL(*child1, OnMouseWheel(_, kDelta, _)).Times(Exactly(1)).WillRepeatedly(Return(false));
```

Comment:
Why not also checking OnMouseWheel for child2 was only executed once?

Reply:
I kind of forgot child2 existed. I'll add this.

Reply:
Oh wait, I can't add this - child2 isn't mocking this method, it's actually returning true. It's intended to test correct return values for the parent (see last `EXPECT_TRUE`), not verify calls to the child.

Reply:
Turns out I misunderstood how gmock works. I've changed the code and I'm now also expecting calls to child2, ptal.